parquet/src/main/java/org/apache/iceberg/parquet/ParquetFileMerger.java

```diff
@@ -0,0 +1,684 @@
+   * @param codec Compression codec to use (should match the file's codec)
+   * @param minValue Minimum value for statistics
+   * @param maxValue Maximum value for statistics
+   * @param valueGenerator Function that generates the value for row i (0-based index)
+   * @throws IOException if writing fails
+   */
+  private static void writeLongColumnChunk(
+      ParquetFileWriter writer,
+      ColumnDescriptor descriptor,
+      long rowCount,
+      CompressionCodecName codec,
+      long minValue,
+      long maxValue,
+      LongUnaryOperator valueGenerator)
+      throws IOException {
+
+    // Start the column chunk with the specified compression codec
+    writer.startColumn(descriptor, rowCount, codec);
+
+    // Use DELTA_BINARY_PACKED encoding for efficient compression
+    ValuesWriter valuesWriter =
+        new DeltaBinaryPackingValuesWriterForLong(
+            DEFAULT_INITIAL_BUFFER_SIZE,
+            DEFAULT_PAGE_SIZE_FOR_ENCODING,
+            HeapByteBufferAllocator.getInstance());
+
+    // Write values using the provided generator function
+    for (long i = 0; i < rowCount; i++) {
+      valuesWriter.writeLong(valueGenerator.applyAsLong(i));
+    }
+
+    // Get the encoded bytes
+    BytesInput encodedData = valuesWriter.getBytes();
+    int uncompressedSize = (int) encodedData.size();
+
+    // Compress the encoded data if a codec is specified
+    BytesInput compressedData;
+    if (codec != CompressionCodecName.UNCOMPRESSED) {
+      CodecFactory codecFactory =
+          CodecFactory.createDirectCodecFactory(
+              new Configuration(), DirectByteBufferAllocator.getInstance(), 0);
+      CodecFactory.BytesCompressor compressor = codecFactory.getCompressor(codec);
+      compressedData = compressor.compress(encodedData);
+    } else {
+      compressedData = encodedData;
+    }
+
+    // Create statistics for the column
+    LongStatistics stats = new LongStatistics();
+    stats.setMinMax(minValue, maxValue);
+    stats.setNumNulls(0);
+
+    // Write data page using DELTA_BINARY_PACKED encoding
+    // For required column (no nulls), we don't need repetition/definition level encoding
+    writer.writeDataPage(
+        (int) rowCount, // valueCount
+        uncompressedSize, // uncompressedSize (size before compression)
+        compressedData, // bytes (compressed encoded data)
+        stats, // statistics
+        Encoding.BIT_PACKED, // rlEncoding (not used for required)
+        Encoding.BIT_PACKED, // dlEncoding (not used for required)
+        Encoding.DELTA_BINARY_PACKED); // valuesEncoding
+
+    // End the column chunk
+    writer.endColumn();
+  }
+
+  /**
+   * Copies all column chunks from a row group using binary copy.
+   *
+   * @param writer ParquetFileWriter to write to
+   * @param baseSchema Base schema without _row_id column
+   * @param inputFile Input file to read from
+   * @param rowGroup Row group metadata
+   * @throws IOException if copying fails
```

Comment:
These don't add much value

Reply:
Removed the parameter descriptions and the @throws tag.